### PR TITLE
[SPARK-11188][SQL] Elide stacktraces in bin/spark-sql for AnalysisExceptions

### PR DESCRIPTION
**Pull request by dilipbiswal**
Only print the error message to the console for Analysis Exceptions in sql-shell

### CODE REVIEW

**Comment by dilipbiswal** [comment]
@marmbrus this is for 1.4.2 branch..

**Comment by marmbrus** [comment]
ok to test

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/AbstractSparkSQLDriver.scala [hunk 1]
```diff
@@ -79,3 +91,6 @@ private[hive] abstract class AbstractSparkSQLDriver(
     tableSchema = null
   }
 }
+
+private[hive] case class CommandProcessorResponseWrapper (rc : CommandProcessorResponse,
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/AbstractSparkSQLDriver.scala]: This is not the right style: https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide

```scala
private[hive] case class CommandProcessorResponseWrapper(
    rc: CommandProcessorResponse,
    cause: Throwable)
```

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 2]
```diff
@@ -276,19 +278,23 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
...
 
-          ret = rc.getResponseCode
+          ret = rcWrapper.rc.getResponseCode
           if (ret != 0) {
-            console.printError(rc.getErrorMessage())
-            driver.close()
-            return ret
+            // For analysis exception, only the error is printed out to the console.
+            rcWrapper.cause match {
+              case e : AnalysisException =>
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala]: no space after `e`.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 3]
```diff
@@ -82,6 +95,15 @@ class CliSuite extends SparkFunSuite with BeforeAndAfter with Logging {
           foundAllExpectedAnswers.trySuccess(())
         }
       }
+      else
+        {
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: no new line.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 4]
```diff
@@ -48,9 +48,22 @@ class CliSuite extends SparkFunSuite with BeforeAndAfter with Logging {
...
+   * @param timeout maximum time for the commands to complete
+   * @param extraArgs any extra arguments
+   * @param errorResponses a sequence of strings whose presence in the stdout of the forked process
+   *                       is taken as an immediate error condition. That is: if a line containing
+   *                       with one of these strings is found, fail the test immediately.
+   *                       The default value is `Seq("Error:")`
+   *
+   * @param queriesAndExpectedAnswers one or more tupes of query + answer
+   */
+
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: no new line

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 5]
```diff
@@ -21,6 +21,8 @@ import scala.collection.JavaConversions._
 
 import java.io._
 import java.util.{ArrayList => JArrayList}
+import org.apache.spark.sql.AnalysisException
+
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala]: no new line

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/AbstractSparkSQLDriver.scala [hunk 6]
```diff
@@ -17,6 +17,8 @@
 
 package org.apache.spark.sql.hive.thriftserver
 
+import org.apache.spark.sql.AnalysisException
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/AbstractSparkSQLDriver.scala]: ordering

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 7]
```diff
@@ -82,6 +95,15 @@ class CliSuite extends SparkFunSuite with BeforeAndAfter with Logging {
...
       }
+      else
+        {
+          errorResponses.foreach { r =>
+            if (line.contains(r)) {
+              foundAllExpectedAnswers.tryFailure(
+                new RuntimeException(s"Failed with error line '$line'"))
+            }
+          }
+        }
```
Comment (marmbrus) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: indent is off due to newline.

**Comment by dilipbiswal** [comment]
@marmbrus hello Michael, even though the failure is in CliSuite, i don't think its related to the change as i have only made cosmetic changes in the last commit. I also ran CliSuite in my 142 env with my changes and am able to run it cleanly. Can we retest this please ?

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
Thanks, merging.  Can you close this?

**Comment by dilipbiswal** [comment]
@marmbrus Thank you very much. Closed the PR.

**Pull request closed without merging**